### PR TITLE
Remove RefreshTimer

### PR DESCRIPTION
**Pull request by sgoldberg-sfdc**
-  This was put in place due to an old defect in NSMutableURLRequest that would not honor the timeout value
-  This defect in NSMutableURLRequest has been fixed back in iOS 6
-  I removed the timer and associated logic, outward behavior should be unchanged.

### CODE REVIEW

**Comment by sgoldberg-sfdc** [comment]
@bhariharan @khawkins @trooper2013

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m [hunk 1]
```diff
@@ -703,17 +699,19 @@ - (void)beginTokenEndpointFlow:(SFOAuthTokenEndpointFlow)flowType {
     NSData *encodedBody = [params dataUsingEncoding:NSUTF8StringEncoding];
     [request setHTTPBody:encodedBody];
     
-    // We set the timeout value for NSMutableURLRequest above, but NSMutableURLRequest has its own ideas
-    // about managing the timeout value (see https://devforums.apple.com/thread/25282).  So we manage
-    // the timeout with an NSTimer, which gets started here.
-    [self startRefreshFlowConnectionTimer];
+    request.timeoutInterval = self.timeout;
```
Comment (khawkins) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m]: Do we even need this now, given that it's set in [Line 650](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/1969/files#diff-a19763c4fa428fe97ccc907d35e442c3R650)?

**Comment by sgoldberg-sfdc** [comment]
build

**Comment by huminzhi** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Code coverage under minimum of 80%</td>
    </tr>
  </tbody>
</table>

## Current coverage for SalesforceSDKCore.framework is `48%`
Files changed | - | - 
--- | --- | ---
SFOAuthCoordinator.m | `32%` | :no_entry_sign:

---
## Current coverage for SalesforceAnalytics.framework is `41%`
✅ *No files affecting coverage found*

---

> Powered by [xcov](https://github.com/nakiostudio/xcov)

<p align="right" data-meta="generated_by_CI-result">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Pull request merged**